sound/soc/sof/control.c

```diff
@@ -416,6 +416,14 @@ int snd_sof_bytes_ext_put(struct snd_kcontrol *kcontrol,
 	if (copy_from_user(&header, tlvd, sizeof(const struct snd_ctl_tlv)))
 		return -EFAULT;
 
+	/* Check that length does not exceed the maximum defined for SOF */
+	if (header.length > SND_SOF_CTL_TLV_LENGTH_MAX) {
+		dev_err_ratelimited(sdev->dev,
+				    "error: Bytes size %d exceeds max %d.\n",
+				    header.length, SND_SOF_CTL_TLV_LENGTH_MAX);
+		return -EINVAL;
```

Comment:
I'm not 100% sure but for this kind of error it might be better to use ENOBUFS, ENOSPC or ENOMEM seem to be less appropriate but might be an option too. Same below.

Reply:
Yep, thanks for suggestion.

Reply:
@singalsu @lyakh I was wondering the same as well, but I think EINVAL is more appropriate as this is exceeding a limit that is part of the interface. ENOMEM/NOSPC are typically for dynamic allocation where the state of the system is such that resource allocation fails. In this case this is a programming error so I think EINVAL is most appropriate.